infrastructure/terraform/accounts/modules/assumable_role/federated/main.tf

```diff
@@ -0,0 +1,37 @@
+resource "aws_iam_role" "role" {
+  name               = var.name
+  assume_role_policy = data.aws_iam_policy_document.assume_role.json
+
+  max_session_duration = var.max_session_duration_in_seconds
+}
+
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Federated"
+      identifiers = [var.federated_principal]
+    }
+
+    actions = ["sts:AssumeRoleWithSAML"]
+
+    condition {
+      test     = "StringEquals"
+      values   = ["https://signin.aws.amazon.com/saml"]
+      variable = "SAML:aud"
+    }
+  }
+
+  # This statement allows you to "double hop" between assumed roles
```

Comment:
Can you provide an example? I think I know what this means, but I'm not 100% sure.